WATOLINK/bci-gaming
# Health bar flash fix

## ClickUp ticket link

Feature fix for #43 [Implement Enemy Health](https://app.clickup.com/t/8686fhu7c)



## Implementation description
Adds a flash to the healthbar when damage is taken. Color of healthbar is assignment in code, can ignore the white healthbar while in the scene view, it returns to normal during gameplay.

Changes to Test New Player:
- Updated the PlayerHealthBar prefab
- Changed the canvas reference size to 1920x1080

If there are any merge conflicts with future branches, can just ignore incoming changes and use the updated prefab on a 1920x1080 reference canvas. 

---

Also, for the enemy health bar, if we want it to not flip when the enemy does, a quick fix would be to create a common parent for both the enemy and health bar:
- Bandit-1-parent
  - Bandit-1
  - Bandit-1-health-group

This wasn't implemented in this branch but just wanted to put this out there.



## Steps to test
1. Go to Assets/Scenes/Test New Player
2. Set your display aspect ratio to Full HD (1920x1080) for best results (I didn't do this before and the healthbar looks pretty weird on low resolutions lol)
3. Play and check that flash on damage works. Can look at the Health Bar (Script) on Canvas/PlayerHealthBar prefab for more options



## What should reviewers focus on?
Does the health bar look good? Changes to location / size can be made, if done also update the prefab!


## Checklist
